test/gtest/include/common/view-tests.hpp

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+TYPED_TEST_P(CommonTests, Subrange) {
+  using DV = typename TypeParam::DV;
+  using V = typename TypeParam::V;
+
+  std::size_t n = 10;
+
+  auto neg = [](auto &v) { v = -v; };
+  DV dv_a(n);
+  TypeParam::iota(dv_a, 100);
+  TypeParam::for_each(TypeParam::policy(),
+                      rng::subrange(dv_a.begin() + 1, dv_a.end() - 1), neg);
+
+  if (comm_rank == 0) {
+    V a(n), a_in(n);
+    rng::iota(a, 100);
+    rng::iota(a_in, 100);
+    rng::for_each(rng::subrange(a.begin() + 1, a.end() - 1), neg);
```

Comment:
We do not test `for_each`. Let's remove its usage from `subrange` tests. See [unittests of subrange](https://github.com/ericniebler/range-v3/blob/master/test/view/subrange.cpp) for examples how to write simple checks on subrange results (check begin/end, dereference).